### ISSUE TITLE
Check should return a more descriptive error message

### ISSUE THREAD

**Issue by juho** [meteor/meteor-feature-requests#199]
`check()` currently returns match errors like:

```
Match error: Expected string, got undefined
```

which don't bubble well. It would be more helpful if it also gave the key it's hanging up on or include a full stack (maybe just in development).

**Comment by mitar** [comment]
Yes, in one project we wrapped those checks into our own check function which then under message encodes a JSON with more details. And then on the other side we parse that.

**Comment by mpowaga** [comment]
For this you would need to know the name of the variable that was passed to `check`. As far as I know it's not possible in javascript (maybe can be retrieved from stack trace).

You can get better error messages by validating an object, e.g.

```javascript
const title = 123;
check({ title }, { title: String });
```

will produce `Match error: Expected string, got number in field title`.

**Comment by mitar** [comment]
Yes, we pass the name of the argument to our check function.

Anyway, the recommended way for method is to accept an object for arguments anyway, instead of position-based arguments. In this way it is easy to extend. Do you know:

https://github.com/meteor/validated-method

**Comment by juho** [comment]
A stack and/or module name would be great. We're running into situations where calling something along the way fails a check but finding which class constructor it was gets hairy when they are initialized with pretty much the same variable names.
